question-servers/freeformPythonLib/prairielearn.py

```diff
@@ -776,8 +772,18 @@ def string_to_2darray(s, allow_complex=True):
         if (m == 0):
             return (None, {'format_error': 'Matrix has no rows.'})
 
-        # Get number of columns by splitting first row on space
-        n = len(s[0].split())
+        # Regex to split rows a la MATLAB
+        matlab_delimiter_regex = re.compile(r'\s*[\s|,]\s*')
```

Comment:
If you are going to use the character set operator `[]` then I think you don't want the alternative operator `|`, because inside `[]` the vertical bar is interpreted as a literal character. For example, the regex as written will also match `4|5|23|2`.
```suggestion
        matlab_delimiter_regex = re.compile(r'\s*[\s,]\s*')
```

Reply:
Oops yeah. I switched brackets because `r'\s*(\s|,)\s*'` was capturing all the delimiters, e.g. `1, 2, 3' -> ['1', ',', '2', ',', '3']`

Reply:
`r'\s*[\s,]\s*'` is working on my end

Reply:
excellent!